mrp_product_document_link/models/mrp_production.py

```diff
@@ -0,0 +1,31 @@
+# © 2025 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See https://www.gnu.org/licenses/lgpl-3.0.html
+
+from odoo import fields, models
+
+
+class MrpProduction(models.Model):
+    _inherit = "mrp.production"
+
+    product_document_urls = fields.Text(
+        string="Documentos",
+        compute="_compute_get_product_document_urls"
+        )
+
+    def _compute_get_product_document_urls(self):
+        for production in self:
+            attachments = self.env["ir.attachment"].search(
+                [
+                    ("res_model", "in", ["product.product", "product.template"]),
+                    ("res_id", "=", production.product_id.id),
+                ]
+            )
```

Comment:
Looking for attachments for both product variant itself and its template is wrong, because product template identifier is different than product one, try this (not tested, it's simply look for the two sets separately and then merge them, please pay attention to `product_tmpl_id` field):
```suggestion
            prod_attachments = self.env["ir.attachment"].search(
                [                
                    ("res_model", "=", "product.product"),
                    ("res_id", "=", production.product_id.id),
                ]
            )
            tmpl_attachments = self.env["ir.attachment"].search(
                [                
                    ("res_model", "=", "product.template"),
                    ("res_id", "=", production.product_id.product_tmpl_id.id),
                ]
            )
            attachments = (prod_attachments | tmpl_attachments)
```